core/iwasm/common/wasm_runtime_common.c

```diff
@@ -348,7 +365,11 @@ wasm_runtime_env_init()
         goto fail5;
     }
 #endif
-
+#ifdef OS_ENABLE_BLOCK_INSN_INTERRUPT
+    if (!os_interrupt_block_insn_init(interrupt_block_insn_sig_handler)) {
+        goto fail6;
```

Comment:
No `os_interrupt_block_insn_destroy` function is defined to destroy the signal, normally the below fails should call it.

Suggest to add `os_interrupt_block_insn_init` and `os_interrupt_block_insn_destroy` into `runtime_signal_init` and `runtime_signal_destroy` to avoid changing this function.

Reply:
What do you mean by "destroy the signal"? For HW bound checks, `thread_signal_inited` is set to false, but in my case I don't see any necessary action to perform https://github.com/bytecodealliance/wasm-micro-runtime/blob/2eed50b3034b8eec531cce4a782b8c698fea00fd/core/shared/platform/common/posix/posix_thread.c#L629

Reply:
OK, I original thought that we should unset the signal handler, e.g. sigaction(.., NULL), but I found the old code also didn't do that.